packages/context/src/__tests__/acceptance/class-level-bindings.acceptance.ts

```diff
@@ -181,6 +181,32 @@ describe('Context bindings - Injecting dependencies of classes', () => {
     return expect(getter()).to.be.fulfilledWith('data');
   });
 
+  it('reports an error if @inject.getter has a non-function target', async () => {
+    ctx.bind('key').to('value');
+
+    class Store {
+      constructor(@inject.getter('key') public getter: string) {}
+    }
+
+    ctx.bind('store').toClass(Store);
+    expect(() => ctx.getSync<Store>('store')).to.throw(
```

Comment:
Nitpick: we can test in an async way`await expect(ctx.get<Store>('store')).to.be.rejectedWith('The type of Store.constructor[0] (String) is not a Getter function')`.